backend/src/utils/agent.py

```
@@ -4,40 +4,30 @@
 from sqlalchemy import text
 from langgraph.errors import NodeInterrupt
 
 from src.repos.thread_repo import ThreadRepo
-from src.entities.a2a import A2AServer
-from src.services.mcp import McpService
+from src.schemas.entities.a2a import A2AServer
 from src.repos.user_repo import UserRepo
-from src.constants import APP_LOG_LEVEL, ARCADE_API_KEY, UserTokenKey
-from src.tools import dynamic_tools
+from src.constants import APP_LOG_LEVEL
+from src.tools import dynamic_tools, init_tools
```

Ruff (0.11.9)

`src.tools.dynamic_tools` imported but unused

Remove unused import: `src.tools.dynamic_tools`

(F401)